src/shared/sam/cli/samCliInvoker.ts

```diff
@@ -31,79 +32,65 @@ interface SamCliProcessInvokerContext {
+        validator: params.validator || defaults.validator,
     }
 }
 
 export class DefaultSamCliProcessInvoker implements SamCliProcessInvoker {
 
-    public constructor(private readonly _context: SamCliProcessInvokerContext) {}
+    public constructor(private readonly _context: SamCliProcessInvokerContext = resolveSamCliProcessInvokerContext()) {}
 
     public invoke(options: SpawnOptions, ...args: string[]): Promise<ChildProcessResult>
     public invoke(...args: string[]): Promise<ChildProcessResult>
     public async invoke(first: SpawnOptions | string, ...rest: string[]): Promise<ChildProcessResult> {
-        const logger: Logger = getLogger()
 
         const samCliLocation = this._context.cliConfig.getSamCliLocation()
         if (!samCliLocation) {
             const err = new Error('SAM CLI location not configured')
-            logger.error(err)
+            this._context.logger.error(err)
             throw err
         }
         const validationResult = await this.getCliValidation(samCliLocation)
 
-        if (validationResult.validation === SamCliVersionValidation.Valid) {
-            const args = typeof first === 'string' ? [ first, ...rest ] : rest
-            const options: SpawnOptions | undefined = typeof first === 'string' ? undefined : first
-
-            return await this.runCliCommand(samCliLocation, options, ...args)
-        }
-
-        const errorResult: ChildProcessResult = {
-            exitCode: 1,
-            stdout: '',
-            stderr: '',
-            error: undefined
-        }
-        switch (validationResult.validation) {
-            case SamCliVersionValidation.VersionTooHigh:
-                const samVersionTooHighMessage = 'AWS Toolkit is out of date'
-                errorResult.error = new Error(samVersionTooHighMessage)
-                errorResult.stdout = samVersionTooHighMessage
-                errorResult.stderr = samVersionTooHighMessage
-                break
-            case SamCliVersionValidation.VersionTooLow:
-            case SamCliVersionValidation.VersionNotParseable:
-            default:
-                const samVersionTooLowMessage = 'SAM CLI is out of date'
-                errorResult.error = new Error(samVersionTooLowMessage)
-                errorResult.stdout = samVersionTooLowMessage
-                errorResult.stderr = samVersionTooLowMessage
-                break
+        if (validationResult.validation !== SamCliVersionValidation.Valid) {
+            // prompt will redirect to external URL for updating. We do not need to wait on this.
+            // tslint:disable-next-line: no-floating-promises
+            this._context.validator.notifyVersionIsNotValid(validationResult)
+            const versionErr = new InvalidSamError(validationResult)
+            this._context.logger.error(
+                versionErr,
+                `${validationResult.validation}: version ${validationResult.version}`
+            )
```

Comment:
Did you mean to throw versionErr or somehow exit out, or do you intend to proceed to call runCliCommand?

Reply:
Yup, I missed it. Just pushed a fix (tests were breaking because of this)